Code/Framework/AzToolsFramework/AzToolsFramework/Viewport/ViewBookmarkLoader.cpp

```diff
@@ -95,7 +97,25 @@ namespace AzToolsFramework
         }
     }
 
-    bool ViewBookmarkLoader::SaveLastKnownLocationInLevel(ViewBookmark bookmark)
+    bool ViewBookmarkLoader::ModifyBookmarkAtIndex(ViewBookmark bookmark, int index, const StorageMode mode)
+    {
+        switch (mode)
+        {
+        case StorageMode::Shared:
+            if (SharedViewBookmarkComponent* bookmarkComponent = FindBookmarkComponent<SharedViewBookmarkComponent>())
+            {
+                return bookmarkComponent->ModifyBookmarkAtIndex(index, bookmark);
+            }
```

Comment:
update: This is not safe I'm afraid as there's the possibility of having more than one `SharedViewBookmarkComponent` in a level (that could be added to several different entities). That's why we want to move some of the logic to be directly inside the `SharedViewBookmarkComponent` (or have it call the same util code)

I'd be tempted to remove this for now and focus exclusively on the `LocalViewBookmarkComponent`

Reply:
This happens throughout this file and might be something to move away from, perhaps the bookmark loader is instance is just for the local bookmarks? Might be something to consider how we handle multiple `SharedViewBookmarkComponent`s

Reply:
Yeah agree, at the beginning I was considering having just one `SharedViewBookmarkComponent` but seeing how things are evolving looks like moving some functionality into the component itself is the most rational solution